gammapy/modeling/models/spectral.py

```diff
@@ -316,13 +319,35 @@ def plot(
 
         energy_min, energy_max = energy_range
         energy = MapAxis.from_energy_bounds(
-            energy_min, energy_max, n_points, energy_unit
-        ).edges
+                                            energy_min, energy_max, n_points, energy_unit
+                                            ).edges
+
+        if sed_type:
+            if sed_type == "dnde":
+                energy_power = 0
```

Comment:
The `energy_power` can remain an independent parameter, no need to define it here...just remove.

Reply:
Ok.